# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



**Описание проекта**

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Описание данных**
* `children` — количество детей в семье
* `days_employed` — общий трудовой стаж в днях
* `dob_years` — возраст клиента в годах
* `education` — уровень образования клиента
* `education_id` — идентификатор уровня образования
* `family_status` — семейное положение
* `family_status_id` — идентификатор семейного положения
* `gender` — пол клиента
* `income_type` — тип занятости
* `debt` — имел ли задолженность по возврату кредитов
* `total_income` — ежемесячный доход
* `purpose` — цель получения кредита

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [107]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [108]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [109]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [110]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [111]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [112]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [113]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [114]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [115]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [116]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [117]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [118]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [119]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [120]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [121]:
data.duplicated().sum()

71

In [122]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [123]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [124]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [125]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [126]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [127]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Узнаем какая часть клиентов с детьми и без детей имеют задолженность по возврату кредита.

In [128]:
# Сгруппируем данные из столбца "наличие задолженностей по возврату кредита" по количеству детей. 
# Выведем среднее значение и количество записей для каждого случая
# Отсортируем по возрастанию среднего значения

debt_children = data.groupby('children')['debt'].agg(['mean', 'count']).sort_values(by = 'mean')

# Представим среднее сначение в процентном выражении
def percent(x):
    return "{0:.2%}".format(x)

debt_children['mean'] = debt_children['mean'].apply(percent)
debt_children

mean  count
children              
5         0.00%      9
0         7.54%  14091
3         8.18%    330
1         9.23%   4808
2         9.45%   2052
4         9.76%     41

**Вывод:** 

- В представленном датафрейме заемщики с 5 детьми не имели задолженности по возврату кредитов. Но их количество слишком мало (9 клиентов из 21525), чтобы делать общие выводы.
- Чаще всего берут кредиты клиенты, не имеющие детей (14091 записей). Эта категория заемщиков менее склонна к невозврату кредитов, чем клиенты, имеющие детей (7,54%).  
- Процент случаев появления задолженнсти по возврату кредита у клиентов с детьми выше (8.18 - 9.76%%).

Однако, прямой зависимости между количеством детей и задолженностью по возврату кредитов не наблюдается.

***Исправлено***

Попробуем разбить клиентов на категории:
- без детей,
- семьи, в которых 1 ребенок,
- семьи, в которых 2 ребенка,
- многодетные семья (3 и более детей).

Посчитаем процент задолженностей по возврату кредитов в этих группах

In [129]:
#Функция сhildren_category принимает каждое значение из столбца children
# и распределяет его в группы соответственно количеству детей
def children_category(children):
    if children == 0:
        return 'нет детей'
    elif children == 1:
        return '1 ребенок'
    elif children == 2:
        return '2 детей'
    return 'многодетная семья'

# добавим в датафрейм столбец children_category, найдем среднее сзачение столбца debt для каждой категории
# отсортируем по возрастанию среднего значения
data['children_category'] = data['children'].apply(children_category)
debt_children_1 = data.groupby('children_category')['debt'].agg(['mean', 'count']).sort_values(by='mean')

# выразим среднее в процентах
def percent(x):
    return "{0:.2%}".format(x)

debt_children_1['mean'] = debt_children_1['mean'].apply(percent)
debt_children_1

mean  count
children_category              
нет детей          7.54%  14091
многодетная семья  8.16%    380
1 ребенок          9.23%   4808
2 детей            9.45%   2052

**Вывод:** 

- 2/3 от всех клиентов - не имеют детей, среди них самый низкий процент задолженности по возврату кредита.
- Многодетныех клиентов менее 2%. Процент задолженности по возврату у них выше клиентов без детей.
- Клиенты, у которых 1 ребенок составляют 1/4 часть выборки, в этой группе процент задолженности по возврату выше остальных.
- Клиенты с 2 детьми имеют самый высокий показатель просроченной задолженности по кредиту. 
- Абсолютная разница между минимальным и максимальным средним значением составляет 20%.

На основании этих данных не удается установить корреляцию между количеством детей у клиента и возвратом кредита в срок. 
Необходимы более точные инструменты для оценки такой зависимости.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Узнаем, какие значения есть в столбце family_status

In [130]:
data['family_status'].value_counts()

женат / замужем          12261
гражданский брак          4134
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
Name: family_status, dtype: int64

In [131]:
# Сгруппируем данные из столбца "наличие задолженностей по возврату кредита" по категориям семейного положения. 
# Выведем среднее значение и количество записей для каждого случая

debt_family_status = data.groupby('family_status')['debt'].agg(['mean', 'count']).sort_values(by='mean')

# выразим среднее в процентах

debt_family_status['mean'] = debt_family_status['mean'].apply(percent)
debt_family_status

mean  count
family_status                      
вдовец / вдова         6.62%    951
в разводе              7.06%   1189
женат / замужем        7.56%  12261
гражданский брак       9.31%   4134
Не женат / не замужем  9.76%   2796

**Вывод:** 
- Группы вдовец/вдова и клиенты в разводе - самая малочисленная (10 часть выборки) и самая ответсвенная к возврату кредита категория клиентов. Объединив их в одну категорию, можно сказать, что люди, раннее состоявшие в отношениях, чаще возвращают кредит в срок.
- Более половины от всех заемщиков состоят в отношениях (официальный брак). В этой категории процент задолженности по возврату кредита выше. 
- Около 1/3 заемщиков не женаты/не замужем или состоят в гражданском браке. Эта категория клиентов имеет самый высокий процент задолженности по возврату кредита. 


Можно сделать вывод о влиянии семейного положения на факт возврата кредита в срок.

In [132]:
debt_family_status1 = pd.pivot_table(data,                                     
                                     columns = 'family_status',
                                     values = ('dob_years', 'debt', 'total_income'),
                                     aggfunc = 'mean'                                     
                                    )
debt_family_status1

family_status  Не женат / не замужем      в разводе  вдовец / вдова  \
debt                        0.097639       0.070648        0.066246   
dob_years                  38.368026      45.561817       56.501577   
total_income           166552.384478  167955.227082   142559.789695   

family_status  гражданский брак  женат / замужем  
debt                   0.093130         0.075606  
dob_years             42.067731        43.558519  
total_income      164819.409289    166757.122584

***Исправлено***

Полагаю, люди, ранее состоявшие в отношениях (группа клиентов со статусом вдовец/вдова и в разводе) - это наиболее возрастная категория клиентов. В силу возраста и накопленного жизненного опыта, в том числе семейных отношений, клиенты этой категории лучше рассчитывают свои возможности, просчитывают риски и осознают свою ответственность, когда решают взять кредит.

Клиенты, состоящие в официальном браке - это пары, где решение о кредите, как правило, принимается совместно. Муж/жена зачастую выступают созаемщиками при заключении кредитного договора, и делят ответственность по погашению кредита. Хотелось бы сказать, что эта группа клиентов так же более финансово стабильна, чем заемщики, не состоящие в официальных отношениях, но по среднему доходу такого вывода сделать нельзя. 

Группа с самым высоким показателем просроченной задолженности - это клиенты не женатые/ не замужем. Вероятно, эта группа моложе остальных, и более "мобильная" и открытая для перемен. Поменять работу, место жительства, принять решение завести семью или получить дополнительное образование - в процессе могут измениться и цели кредита и возможности его своевременного погашения. 

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Узнаем (ради интереса) максимальное, минимальное, среднее и медианное значения в столбце 'total_income':

In [133]:
data['total_income'].agg(['min','max','mean','median']).apply(lambda x: '%.4f' % x)

min         20667.0000
max       2265604.0000
mean       165342.7458
median     142594.0000
Name: total_income, dtype: object

Клиенты по суммарному доходу разбиты на 5 категорий:
   - A - клиенты с доходом свыше 1 000 000  руб.
   - B - клиенты с доходом от 200 001 до 1 000 000 руб.
   - C - клиенты с доходом от 50 001 до 200 000 руб.
   - D - клиенты с доходом от 30 001 до 50 000 руб.
   - E - клиенты с доходом до 30 000 руб.

Узнаем процент задолженностей по возврату кредитов для каждой группы:

In [135]:
# Сгруппируем данные из столбца "наличие задолженностей по возврату кредита" по категориям дохода. 
# Выведем среднее значение и количество записей для каждого случая

debt_income = data.groupby('total_income_category')['debt'].agg(['mean', 'count']).sort_values(by='mean')

#Выразим среднее значение в процентах

debt_income['mean'] = debt_income['mean'].apply(percent)
debt_income

mean  count
total_income_category              
D                      6.02%    349
B                      7.06%   5014
A                      8.00%     25
C                      8.50%  15921
E                      9.09%     22

**Вывод:** 

Группы с самым низким и с самым высоким доходом - малочисленные (22 и 25 записей соответственно). Если принять эти значения за экстремальные и не рассматривать их в этом срезе, то можно сделать следующие выводы:
- Около 3/4 заемщиков принадлежат к группе с доходом 50 001 - 200 000 руб. Процент задолженности по возврату кредита в этой группе самый высокий среди всех показателей. 
- При этом в группе клиентов с доходами 200 001 - 1 000 000 руб., второй по численности (1/4 от общего количества), процент задолженности ниже.
- Лучший показатель по своевременному возврату кредита у клиентов с доходом 30 000 - 50 000 руб.

Таким образом, прямой зависимости между уровнем дохода клиента и возвратом кредита в срок не наблюдается. Но стоит обращать внимание на принадлежность клиента к той или иной категории.


#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [136]:
# Сгруппируем данные из столбца "наличие задолженностей по возврату кредита" по целям кредита. 
# Выведем среднее значение и количество записей для каждого случая

debt_purpose = pd.pivot_table(data, 
                             index = 'purpose_category',
                             values = 'debt', 
                             aggfunc = ['mean', 'count'],
                             )

debt_purpose[('mean', 'debt')] = debt_purpose[('mean', 'debt')].apply(percent) 

debt_purpose.sort_values(by=[('mean', 'debt')])

mean  count
                           debt   debt
purpose_category                      
операции с недвижимостью  7.26%  10751
проведение свадьбы        7.91%   2313
получение образования     9.25%   3988
операции с автомобилем    9.35%   4279

**Вывод:** 
- Половина всех заемщиков берет кредит для операций с недвижимостью. Эта группа клиентов наиболее ответственно подходит к возврату кредита в срок. На мой взгляд, это связано со сложностью получения ипотечных кредитов и санкциями при невозврате в срок, влекущими потенциальную потерю объекта залога. 
- Среди клиентов, берущих кредит для проведения свадьбы, также невысокий процент задолженности по возврату.
- Повышенный процент задолженности по возврату у клиентов, берущих кредит на операции с автомобилем и для получения образования.


#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 
В исходных данных наблюдались пропуски в столбцах 'days_employed' - 'общий трудовой стаж ' и 'total_income' - 'ежемесячный доход'.
Причины, по которым возникли пропуски в этих столбцах могут быть следующими:
- Техническая ошибка при загрузке/выгрузке данных.
- Отсутствие информации от клиента (например пропуски при заполнении анкеты) или нежелание клиентов предоставлять данные сведения.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 

Медиана отражает центр распределения данных. Она менее чувствительна к экстремальным значениям. Заполнение пропусков медианным значением позволяет не терять данные при предобработке.

### Шаг 4: общий вывод.

В процессе исследования были реализованы следующие действия:

- Импортированы и изучены данные.
- В прцессе предобработки были устранены дубликаты, заполнены пропуски, удалены аномальные значения.
- Уровень дохода клиента и цели кредита разбиты на категории для анализа.
- Проведен анализ влияния семейного положения, количества детей у клиента, уровня дохода и цели получения кредита на факт погашения кредита в срок.

Анализ влияния семейного положения выявил следующую тенденцию: 
* клиенты, раннее состоявшие в отношениях (текущий статус 'вдовец/вдова' и 'в разводе'), являются самой надежной категорией заемщиков. 
* не выявлена взаимосвязь между наличием и количеством детей у клиента и появлением просроченной задолженности.

Категоризация клиентов по уровню дохода показала следующий результат:
* большинство заемщиков относится к группе с доходом 50 000 - 200 000 руб. В этой группе также высокий процент задолженности по своевременному погашению кредита. 
* Большая группа клиентов имеет доход от 200 000 до 1 000 000 руб. и лучшие показатели по своевременному погашению кредита.  Возможно, стоит применить более точные критерии категоризации клиентов или иные методы оценки корреляции уровня дохода и наличия просроченной задолженности. 

Анализ влияния цели кредита на его возврат в срок показал:
* чаще всего берутся, а также своевременно возвращаются кредиты на операции с недвижимостью. В то время как кредиты на образование и приобретение автомобиля имеют больший процент просрочек при погашении.

На мой взгляд, стоило бы так же проанализировать средний возраст клиента, уровень образования, тип занятости и трудовой стаж. 

Тогда к образу идеального заемщика я бы добавила:
- отсутствие детей (и вообще иждивенцев); 
- официально оформленный брак (текущий или в анамнезе); 
- доход от 200 000 до 1 000 000 руб (хотя эта категория требует дополнительной разбивки и уточнения);
- цели кредита - операции с недвижимостью или проведение свадьбы.